File: 322-coin-change/coin-change.java
class Solution {
    public int coinChange(int[] coins, int amount) {
       int[]arr=new int[amount+1];
       for(int x=1;x<arr.length;x++)
       {
        arr[x]=Integer.MAX_VALUE;
        for(int coin:coins)
        {
            if(x>=coin&&arr[x-coin]!=Integer.MAX_VALUE)
            {
                arr[x]=Math.min(arr[x],arr[x-coin]+1);
            }
        }
       }
       if(arr[amount]==Integer.MAX_VALUE) return -1;
       else return arr[amount];
    }
}

File: 322-coin-change/coin-change.js
/**
 * @param {number[]} coins
 * @param {number} amount
 * @return {number}
 */
var coinChange = function(coins, amount) {
    let arr = new Array(amount+1);
    arr[0] = 0;
    let max = Infinity;
    for(let x = 1; x <= amount; x++) {
        arr[x] = max;
        for(let coin of coins) {
            if(x >= coin && arr[x-coin] != Infinity) {
                arr[x] = Math.min(arr[x], arr[x-coin]+1) 
            }
        }
    }
    if(arr[amount] === Infinity) return -1;
    else return arr[amount];
};